packages/ckeditor5-dev-changelog/src/utils/providenewversionformonorepository.ts

```diff
@@ -0,0 +1,148 @@
+ * For licensing, see LICENSE.md.
+ */
+
+import semver, { type ReleaseType } from 'semver';
+import chalk from 'chalk';
+import inquirer from 'inquirer';
+import { randomUUID } from 'crypto';
+import upath from 'upath';
+import os from 'os';
+import fs from 'fs-extra';
+import pacote from 'pacote';
+
+const CLI_INDENT_SIZE = 3;
+
+type Options = {
+	packageName: string;
+	version: string;
+	bumpType: ReleaseType;
+	indentLevel?: number;
+};
+
+type VersionValidationResult = string | true;
+
+type Question = {
+	type: 'input';
+	name: 'version';
+	default: string;
+	message: string;
+	filter: ( input: string ) => string;
+	validate: ( input: string ) => Promise<VersionValidationResult>;
+	prefix: string;
+};
+
+const manifest = cacheLessPacoteFactory( pacote.manifest );
+
+/**
+ * Handles version management for monorepository packages.
+ * This function manages versioning for multiple packages in a monorepository structure.
+ */
+export async function provideNewVersionForMonorepository( options: Options ): Promise<string> {
+	const question = createVersionQuestion( options );
+	const answers = await inquirer.prompt<{ version: string }>( question as any );
+	return answers.version;
+}
+
+/**
+ * Validates if the provided version is valid according to semver.
+ */
+function validateVersionFormat( version: string ): VersionValidationResult {
+	if ( !semver.valid( version ) ) {
+		return 'Please provide a valid version.';
+	}
+	return true;
+}
+
+/**
+ * Validates if the provided version is higher than the current version.
+ */
+function validateVersionHigherThanCurrent( version: string, currentVersion: string ): VersionValidationResult {
+	if ( !semver.gt( version, currentVersion ) ) {
+		return `Provided version must be higher than "${ currentVersion }".`;
+	}
+	return true;
+}
+
+/**
+ * Validates if the provided version is available in the npm registry.
+ */
+async function validateVersionAvailability( version: string, packageName: string ): Promise<VersionValidationResult> {
+	const isAvailable = await checkVersionAvailability( version, packageName );
+	if ( !isAvailable ) {
+		return 'Given version is already taken.';
+	}
+	return true;
+}
+
+/**
+ * Creates a prompt question for version input with validation.
+ */
+function createVersionQuestion( options: Options ): Array<Question> {
+	const { version, packageName, bumpType, indentLevel = 0 } = options;
+	const suggestedVersion = semver.inc( version, bumpType ) || version;
+	const message = 'Type the new version ' +
+		`(current highest: "${ version }" for "${ chalk.underline( packageName ) }", suggested: "${ suggestedVersion }"):`;
+
+	return [ {
+		type: 'input',
+		name: 'version',
+		default: suggestedVersion,
+		message,
+		filter: ( input: string ) => input.trim(),
+		async validate( input: string ): Promise<VersionValidationResult> {
+			const formatValidation = validateVersionFormat( input );
+			if ( formatValidation !== true ) {
+				return formatValidation;
+			}
```

Comment:
1. This validation prevents from typing `internal` as expected version.
2. Please add a blank line before all `if` statements in all files to be consistent.

Reply:
Added in [675242d](https://github.com/ckeditor/ckeditor5-dev/pull/1090/commits/675242db17f8aac2ca8c53dddfea792df1a6ce69).